pages/docs/reference/fun-interfaces.md

```diff
@@ -0,0 +1,101 @@
+---
+type: doc
+layout: reference
+title: "Functional interfaces (SAM interfaces)"
+---
+
+# Functional (SAM) interfaces
+
+An interface with only one abstract method is called a _functional interface_, or a _Single Abstract 
+Method (SAM) interface_.
+
+Functional interfaces in Kotlin are marked with the `fun` keyword.
+
+<div class="sample" markdown="1" theme="idea" data-highlight-only>
+
+```kotlin
+fun interface Action {
+    fun run()
+}
+```
+
+</div>
+
+## SAM conversions
+
+For functional interfaces, you can use SAM conversions that help you make your code more concise and readable thanks to 
+lambda expressions.
+
+Instead of creating a class that implements a functional interface manually, you can use a lambda expression. 
+Thanks to a SAM conversion, Kotlin can transparently convert any lambda expression whose signature matches 
+the signature of the interface's single method into an instance of a class that implements the interface.
+
+For example, consider the following Kotlin functional interface:
```

Comment:
I think this sentence also can be omitted (I have a bad habit putting it into my docs :))

Reply:
Why do you think it's a bad habit? It's good to explain what you do. If I just put a code snippet after generic explanation, it would look strange. Docs should be easy to read and understand. We are not limited in a number of words but all words should be helpful.